File: scala/scala-spring-boot/src/main/scala/com/meridal/examples/scalaspringboot/model/Recording.scala
package com.meridal.examples.scalaspringboot.model

import javax.persistence._

import scala.beans.BeanProperty

@Entity
class Recording(
    @BeanProperty var artist: String = "", 
    @BeanProperty var title: String = "", 
    @BeanProperty var catalogueNumber: String = "") {

    def this() = this("", "", "")
  
    @Id
    @GeneratedValue
    @BeanProperty
    var id: Long = _

    override def equals(other: Any): Boolean = other match {
      case recording: Recording => this.id == recording.id
      case _ => false
    }

    override def hashCode(): Int = if (id eq null) 0 else id.toInt
}

File: kotlin/kotlin-spring-boot-mysql/src/main/kotlin/com/meridal/examples/kotlinspringboot/model/Recording.kt
package com.meridal.examples.kotlinspringboot.model

import java.time.LocalDateTime
import javax.persistence.*

@Entity
@Table(name = "recording")
class Recording (
    var artist: String,
    var title: String,
    var catalogueNumber: String,
    var year: Int,
    @OneToMany(cascade = [CascadeType.ALL], fetch = FetchType.EAGER, orphanRemoval = true)
    @JoinColumn(name = "recording_id")
    var songs: MutableList<Song> = mutableListOf<Song>(),
    @Id @GeneratedValue
    var id: Long = 0L
    ) {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Recording

        if (id == 0L || id != other.id) return false

        return true
    }

    override fun hashCode(): Int = id.hashCode()
}
